development/tools/cmd/synchronizer/main.go

```diff
@@ -0,0 +1,171 @@
+		return
+	}
+
+	messages := make([]t.Message, alertAmount)
+	for i := range reports {
+		messages[i] = reports[i]
+	}
+	err = t.SendMessage(slackToken, strings.Trim(slackChannel, "#"), messages)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+func findComponents(dir string, storage *ComponentStorage) {
+	runner := func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// skip walk if the file is dir or name is not equal to 'pathToVersionFile' value
+		if info.IsDir() || info.Name() != pathToVersionFile {
+			return nil
+		}
+		// skip walk if path has `vendor` dir inside
+		if strings.Contains(path, "/vendor/") {
+			log.Printf("Path %q was skipped beacuse it is 'vendor' path", path)
+			return nil
+		}
+
+		componentDir, err := filepath.Abs(filepath.Dir(path))
+		if err != nil {
+			return errors.Wrap(err, "while finding directory for pathToVersionFile")
+		}
+
+		versionPaths, err := findVersionPath(componentDir)
+		if err != nil {
+			return errors.Wrap(err, "while finding path to version")
+		}
+		// skip walk if path to values.yaml is empty, component has no info about version
+		if len(versionPaths) == 0 {
+			log.Printf("File %s in %q has no %q command", pathToVersionFile, path, varsionPathCommand)
+			return nil
+		}
+
+		synComponent := sc.NewSynComponent(sc.RelativePathToComponent(dir, componentDir), versionPaths)
+		storage.AddComponent(synComponent)
+		return nil
+	}
+
+	if err := filepath.Walk(dir, runner); err != nil {
+		log.Fatalf("Cannot walk for %q directory: %s", dir, err)
+	}
+}
+
+func fillComponentStorage(dir string, storage *ComponentStorage, expiryDays int) {
+	for _, component := range storage.components {
+		// find component git hash
+		hash, err := t.FindCommitHash(dir, component.Path)
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+		component.GitHash = hash
+
+		// find component date commit
+		date, err := t.FindCommitDate(dir, component.Path)
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+		component.CommitDate = date
+
+		// find component versions
+		err = t.FindComponentVersion(dir, component)
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+
+		// find modified files beetwen versions and hash
+		for _, version := range component.Versions {
+			files, err := t.FindFileDifference(dir, component.Path, component.GitHash, version.Version)
+			if err != nil {
+				log.Fatal(err.Error())
+			}
+			version.ModifiedFiles = files
+		}
+		component.SetOutOfDate(expiryDays)
+	}
+}
+
+// fetch data about paths to values.yaml files where version of component exist
+func findVersionPath(dir string) ([]string, error) {
+	cmd := exec.Command("bash", "-c", "make "+varsionPathCommand)
+	cmd.Dir = dir
+
+	res, err := cmd.CombinedOutput()
+	response := string(res)
+	if err == nil {
+		response = strings.Trim(response, "\n")
+		return strings.Split(response, "\n"), nil
+	}
+
+	re := regexp.MustCompile("No rule to make target .*" + varsionPathCommand + ".*")
```

Comment:
why we accept that makefile does not contain required rule?

Reply:
I assume not all parts of Kyma which contains Makefile has to have version in resources of Kyma (maybe some tools has Makefile but is not use as chart in Kyma resources). Logs can shows us it is oversight or specific component doesn't need it.